src/Antiforgery/src/Internal/DefaultClaimUidExtractor.cs

```diff
@@ -134,9 +135,13 @@ private byte[] ComputeSha256(IEnumerable<string> parameters)
 
                 writer.Flush();
 
-                var sha256 = serializationContext.Sha256;
-                var stream = serializationContext.Stream;
-                var bytes = sha256.ComputeHash(stream.ToArray(), 0, checked((int)stream.Length));
+                bool success = serializationContext.Stream.TryGetBuffer(out ArraySegment<byte> buffer);
+                if (!success)
+                {
+                    throw new InvalidOperationException();
+                }
+
+                var bytes = SHA256.HashData(buffer);
```

Comment:
Implicit span conversion?

Reply:
Yes. `ArraySegment<byte>` is implicitly convertible to `ROS<byte>`.